routes/api/follows.js

```diff
@@ -1,152 +1,206 @@
+                  res.json({
+                    success: true,
+                    message: "Followed"
+                  })
+                );
               });
-
-          })
-          .catch(err => res.status(500).json({ message: err.msg }));
-      } else {//no user found
-        return res.status(404).json({ success: false, message: 'There is no such profile' });
-      }
-    })
-    .catch(err => {
-      return res.status(500).json({ success: false, message: err.message });
-    });
-});
+            })
+            .catch(err => res.status(500).json({ message: err.msg }));
+        } else {
+          //no user found
+          return res
+            .status(404)
+            .json({ success: false, message: "There is no such profile" });
+        }
+      })
+      .catch(err => {
+        return res.status(500).json({ success: false, message: err.message });
+      });
+  }
+);
 
 // @route   put api/follow/:userId
 // @desc    Unfollow User
 // @input   Get Userid from req params of person whom user wants to unfollow.
 // @access  Private
-
-router.put('/:userId/unfollow', passport.authenticate('jwt', { session: false }), (req, res) => {
-
-  const userId = req.params.userId;
-  // check if your id doesn't match the id of the user you want to follow
-  if (req.user.id === userId) {
-    return res.status(400).json({ alreadyfollow: "You cannot unfollow follow yourself" })
-  }
-  User.findById(userId)
-    .then(user => {
-      if (user) {//user exists
-        // check if the requested user is already in follower list then remove 
-        if (user.followers.filter(follower =>
-          follower.user.toString() === req.user.id).length > 0) {
-          const followerIndex = user.followers.map(follower => 
-            follower.user.toString().indexOf(req.userid));
-          user.followers.splice(followerIndex, 1);
-          user.save()
-            .then(data => {
-              User.findOne({ email: req.user.email })
-                .then(doc => { //check if unfollow user in following [] then remove
-                  if (doc.following.filter(follow =>
-                    follow.user.toString() === userId).length > 0) {
-                    const followingIndex = doc.following.map(follow =>
-                      follow.user.toString().indexOf(req.userid));
-                    doc.following.splice(followingIndex, 1);
-                    doc.save();
-                    return res.json({
-                      success: true,
-                      message: "Unfollowed"
-                    });
-                  } else {
-                    return res.status(400).json({ message: "You are not the following the user yet" });
-                  }
-                })
+router.put(
+  "/:user_id/unfollow",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    // check if your id doesn't match the id of the user you want to follow
+    if (req.user.id === req.params.user_id) {
```

Comment:
I added const userId = req.params.userId; in PR 19. And every where I used that userId instead of directly using req.params.userId
I did it for Both follow and unfollow. That is missing here. If there are merge conflicts, then retain master code for follow and unfollow

Reply:
okay, when I merge I will keep `const userId = req.params.userId;`

Reply:
ok